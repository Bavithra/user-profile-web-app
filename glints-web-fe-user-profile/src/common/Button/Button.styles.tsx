import styled from "styled-components";
import { Colors } from "../../styles/Colors";

type ButtonProps = {
  isDisabled: boolean;
};

export const Button = styled.button<ButtonProps>`
  color: white;
  background: ${(props) => (props.isDisabled ? Colors.grey6 : Colors.green1)};
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid
    ${(props) => (props.isDisabled ? Colors.grey6 : Colors.green1)};
  border-radius: 3px;
`;
