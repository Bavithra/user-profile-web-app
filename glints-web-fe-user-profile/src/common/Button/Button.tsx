import React from "react";

import { Button as StyledButton } from "./Button.styles";

type Props = {
  children: React.ReactNode;
  isDisable?: boolean;
  onClick?: () => void;
};

export default function Button(props: Props) {
  const { children, isDisable = false, onClick } = props;

  return (
    <StyledButton disabled={isDisable} isDisabled={isDisable} onClick={onClick}>
      {children}
    </StyledButton>
  );
}
