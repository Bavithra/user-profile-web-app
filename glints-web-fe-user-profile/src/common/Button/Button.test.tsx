import React from "react";
import Button from "./Button";
import renderer from "react-test-renderer";

describe("Button Block", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Button children={""} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
