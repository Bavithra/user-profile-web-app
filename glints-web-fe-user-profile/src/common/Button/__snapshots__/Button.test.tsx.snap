// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Block renders correctly 1`] = `
<button
  className="sc-bczRLJ fBHdyQ"
  disabled={false}
/>
`;
